airflow/cli/commands/dag_command.py

```diff
@@ -214,14 +215,37 @@ def dag_unpause(args) -> None:
 @providers_configuration_loaded
 def set_is_paused(is_paused: bool, args) -> None:
     """Set is_paused for DAG by a given dag_id."""
-    dag = DagModel.get_dagmodel(args.dag_id)
-
-    if not dag:
-        raise SystemExit(f"DAG: {args.dag_id} does not exist in 'dag' table")
+    should_apply = True
+    dags = [
+        dag
+        for dag in get_dags(args.subdir, dag_id=args.dag_id, use_regex=args.treat_dag_as_regex)
```

Comment:
I believe dag_id or treat_dag_as_regex args one of them should be set not both. Can we add notification for the same?

Reply:
`--treat-dag-as-regex` is a flag that indicates that `dag_id` should be treated as a regex.
If we consolidate them into one arg (meaning that `dag_id` is always a regex), then `dag_id="dag"` could match `dag`, but also `dag-1`, `dag-2`, `dag-3`, etc., and it will become an unnecessary breaking change.